crates/nu-cli/tests/completions/mod.rs

```diff
@@ -837,8 +837,8 @@ fn subcommand_completions(mut subcommand_completer: NuCompleter) {
     match_suggestions(
         &vec![
             "foo bar".to_string(),
-            "foo aabcrr".to_string(),
             "foo abaz".to_string(),
+            "foo aabcrr".to_string(),
```

Comment:
I don't have a clue why these results changed order. I looked at the scores and this new order is the right one. No idea how the tests were passing before.

Reply:
That was one thing I considered, but Cargo.lock hasn't changed, so I guess the `fuzzy_matcher` crate itself hasn't changed versions

Reply:
Figured this out finally. It's because this PR changes it so instead of sorting suggestions at the end, it uses the same partial string used to match subcommands to also sort them. This means that when sorting `"foo bar"` and `"foo abar"`, they'll be compared to `"foo b"`. Previously, they would've been compared to the provided prefix, which is just `"b"`. I think this change in behavior makes more sense, to be honest. I don't think it's a big enough change to be worth adding a test for, so I think this PR is now ready to be reviewed/merged again.